### PR TITLE
[PLGN-325] RSS - `Poll`: Fixed issue where trigger would return all entries on startup

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - `Poll`: Fixed issue where trigger would return all entries on startup
  - Caching locally in file content is replaced to be stored in memory. Maximum limitation of response in `feed.parse` to default 1000 allows to limit the max memory usage during trigger runtime, as we don't need to store all of the entries. Currently the logic works to store max of 1000 latest entries pulled, compare with new response on each call, if there are new entries found - trigger the workflow for each.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Comment by igorski-r7** [comment]
The `PluginValidator` is failing due the plugin doesn't contain action field in it's `plugin.spec.yaml` file. The `KeyError` is being raised during validation.

File: plugins/rss/help.md [hunk 1]
```diff
@@ -61,133 +63,79 @@ Example input:
...
-_This plugin does not contain any custom output types._
+### Custom Types
+  
+*This plugin does not contain any custom output types.*
 
 ## Troubleshooting
-
-Between workflow runs, new items will *not* be reported on.
+  
+*There is no troubleshooting for this plugin.*
```
Comment (cmcnally-r7) [inline plugins/rss/help.md:134]: Do we still need this? The tooling doesn't support troubleshooting generation atm so it usually clears what already exists
```suggestion
Between workflow runs, new items will *not* be reported on.
```

File: plugins/rss/komand_rss/connection/connection.py [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+        self.feed_url = None
 
-    def connect(self, params):
-        self.FEED_URL = params.get("url")
+    def connect(self, params={}) -> None:
+        self.feed_url = params.get(Input.URL, "")
+
+    def test(self) -> Dict[str, bool]:
+        if not validators.url(self.feed_url):
+            raise ConnectionTestException(
```
Comment (cmcnally-r7) [inline plugins/rss/komand_rss/connection/connection.py:20]: Are there any basic api calls we can make with this service to test for 200 ?

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

File: plugins/rss/help.md (hunk 1 above)
Comment (igorski-r7) [reply plugins/rss/help.md:134]: You are right. Didn't notice so. I'll update it.

File: plugins/rss/komand_rss/connection/connection.py (hunk 2 above)
Comment (igorski-r7) [reply plugins/rss/komand_rss/connection/connection.py:20]: Yup, I can try to fetch entries there. Good catch. I'll extend it 😄

**Review by cmcnally-r7 — APPROVED** [review]

**Review by dlaverty-r7 — APPROVED** [review]
@igorski-r7 lgtm but can you confirm if the validator failure is expected?

**Pull request merged**